src/util/urlUtils.js

```diff
@@ -48,10 +48,17 @@ export function isExcludedUrl(urlArray, route) {
  */
 export function getFullUrl(base, args) {
 	if (!args || Object.keys(args).length === 0) return base;
+
+	// remove hash portion of url if present
```

Comment:
Added functionality and additional tests. Function now supports args that are empty/null/undefined. These are skipped in the generation of the query params. Function also now supports hashes. For example `kiva.org#updates` should have query params inserted before the hashes, and produce something like `kiva.org?query1=true#updates`